modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/DDMStructureLocalServiceTest.java

```diff
@@ -416,6 +490,29 @@ public void testGetFullHierarchyDDMFormFieldsMap() throws Exception {
 			childFullHierarchyDDMFormFieldsMap.containsKey("Description"));
 	}
 
+	@Test
+	public void testGetStructureByExternalReferenceCode() throws Exception {
+		DDMStructure ddmStructure = _addStructure(
+			RandomTestUtil.randomString(), RandomTestUtil.randomString());
+
+		Assert.assertEquals(
+			ddmStructure,
+			_ddmStructureLocalService.getStructureByExternalReferenceCode(
+				ddmStructure.getExternalReferenceCode(),
+				ddmStructure.getGroupId(), ddmStructure.getClassNameId()));
+
+		try {
+			_ddmStructureLocalService.getStructureByExternalReferenceCode(
+				RandomTestUtil.randomString(), ddmStructure.getGroupId(),
+				ddmStructure.getClassNameId());
+
+			Assert.fail();
+		}
+		catch (NoSuchStructureException noSuchStructureException) {
+			Assert.assertNotNull(noSuchStructureException);
+		}
```

Comment:
Brian does not like this way of asserting a failure. He prefers the use of AssertUtils.assertFailure().
Can you try to use it? You can see the usage of this in the portal;